sway/desktop/xwayland.c

```diff
@@ -2,42 +2,94 @@
 #include <stdbool.h>
 #include <stdlib.h>
 #include <wayland-server.h>
-#include <wlr/xwayland.h>
 #include <wlr/types/wlr_output_layout.h>
 #include <wlr/types/wlr_output.h>
+#include <wlr/xwayland.h>
+#include "log.h"
+#include "sway/desktop.h"
+#include "sway/input/input-manager.h"
+#include "sway/input/seat.h"
+#include "sway/output.h"
+#include "sway/server.h"
 #include "sway/tree/container.h"
 #include "sway/tree/layout.h"
-#include "sway/server.h"
 #include "sway/tree/view.h"
-#include "sway/output.h"
-#include "sway/input/seat.h"
-#include "sway/input/input-manager.h"
-#include "log.h"
+
+static void unmanaged_handle_request_configure(struct wl_listener *listener,
+		void *data) {
+	struct sway_xwayland_unmanaged *surface =
+		wl_container_of(listener, surface, request_configure);
+	struct wlr_xwayland_surface *xsurface = surface->wlr_xwayland_surface;
+	struct wlr_xwayland_surface_configure_event *ev = data;
+	wlr_xwayland_surface_configure(xsurface, ev->x, ev->y,
+		ev->width, ev->height);
+}
+
+static void unmanaged_handle_commit(struct wl_listener *listener, void *data) {
+	struct sway_xwayland_unmanaged *surface =
+		wl_container_of(listener, surface, commit);
+	struct wlr_xwayland_surface *xsurface = surface->wlr_xwayland_surface;
+	desktop_damage_from_surface(xsurface->surface, xsurface->x, xsurface->y);
+	// TODO: handle window motion
```

Comment:
:<

Reply:
This reason it's a TODO is that nobody moves an unmanaged window, so it's difficult to test.

Reply:
Can we at least do an implementation that might work, then?

Reply:
ack